getSimpleHeaderValue NullPointerException

Since around one week I am getting the following NPE. It occurs only sporadically. I have changed the way I used the lib and upgraded to 2.12 at the same time, so I am not sure if the issue would occur with previous versions as well. Also it probably might be an issue with DigitalOcean itself, but in that case maybe could be handled nicer in the lib. I have seen it on the following methods: getActionInfo(),  getAllDropletSnapshots(), getDropletInfo(). Here is  a stacktrace:

```
00:00:05 Caused by: java.lang.NullPointerException
00:00:05 	at com.myjeeva.digitalocean.impl.DigitalOceanClient.getSimpleHeaderValue(DigitalOceanClient.java:1749)
00:00:05 	at com.myjeeva.digitalocean.impl.DigitalOceanClient.getSimpleHeaderValue(DigitalOceanClient.java:1761)
00:00:05 	at com.myjeeva.digitalocean.impl.DigitalOceanClient.appendRateLimitValues(DigitalOceanClient.java:1718)
00:00:05 	at com.myjeeva.digitalocean.impl.DigitalOceanClient.executeHttpRequest(DigitalOceanClient.java:1584)
00:00:05 	at com.myjeeva.digitalocean.impl.DigitalOceanClient.doGet(DigitalOceanClient.java:1532)
00:00:05 	at com.myjeeva.digitalocean.impl.DigitalOceanClient.perform(DigitalOceanClient.java:1499)
00:00:05 	at com.myjeeva.digitalocean.impl.DigitalOceanClient.getAllDropletSnapshots(DigitalOceanClient.java:1265)
00:00:05 	at ....

```

--- a/src/main/java/com/myjeeva/digitalocean/impl/DigitalOceanClient.java
+++ b/src/main/java/com/myjeeva/digitalocean/impl/DigitalOceanClient.java
@@ -1715,11 +1715,17 @@ public class DigitalOceanClient implements DigitalOcean, Constants {
       return StringUtils.EMPTY;
     }
 
-    String rateLimitData =
-        String.format(RATE_LIMIT_JSON_STRUCT, getSimpleHeaderValue(HDR_RATE_LIMIT, httpResponse),
-            getSimpleHeaderValue(HDR_RATE_REMAINING, httpResponse),
-            getDateString(getSimpleHeaderValue(HDR_RATE_RESET, httpResponse), DATE_FORMAT));
+    // Occasionally the DigitalOcean API will fail to send rate limit headers.
+    // Simply omit rate limit data in that case.
+    String rateLimit = getSimpleHeaderValue(HDR_RATE_LIMIT, httpResponse);
+    String rateRemaining = getSimpleHeaderValue(HDR_RATE_REMAINING, httpResponse);
+    String rateReset = getSimpleHeaderValue(HDR_RATE_RESET, httpResponse);
+    if (rateLimit == null || rateRemaining == null || rateReset == null) {
+      return response;
+    }
 
+    String rateLimitData = String.format(RATE_LIMIT_JSON_STRUCT,
+      rateLimit, rateRemaining, getDateString(rateReset, DATE_FORMAT));
     log.debug("RateLimitData:: {}", rateLimitData);
 
     return StringUtils.substringBeforeLast(response, "}") + ", " + rateLimitData + "}";
@@ -1740,18 +1746,20 @@ public class DigitalOceanClient implements DigitalOcean, Constants {
    * Easy method for HTTP header values (first/last)
    */
   private String getSimpleHeaderValue(String header, HttpResponse httpResponse, boolean first) {
-    String value = StringUtils.EMPTY;
     if (StringUtils.isBlank(header)) {
-      return value;
+      return StringUtils.EMPTY;
     }
 
+    Header h;
     if (first) {
-      value = httpResponse.getFirstHeader(header).getValue();
+      h = httpResponse.getFirstHeader(header);
     } else {
-      value = httpResponse.getLastHeader(header).getValue();
+      h = httpResponse.getLastHeader(header);
     }
-
-    return value;
+    if (h == null) {
+      return null;
+    }
+    return h.getValue();
   }
 
   /**

